sharedbuffers/mapped_struct.py

```diff
@@ -1719,6 +1738,403 @@ def __str__(self):
+                        elif h1 < h2 or j >= seqlen:
+                            return False
+                        else:
+                            # pfset[j] < val, skip as much as we can
+                            while j < seqlen:
+                                if _stable_hash(pfset.objlist._c_getitem(j, dcode2, objlen2, itemsize2, offset2, _struct2, None)) >= h1:
+                                    break
+                                j += 1
+                            if j == seqlen:
+                                return False
+            return not strict_subset or xlen < seqlen
+        elif xlen > len(seq):
+            return False
+        elif self.objlist is None:
+            if self.bitrep_lo:
+                for i in xrange(64):
+                    if not self.bitrep_lo >> i:
+                        break
+                    elif (self.bitrep_lo & (1 << i)) and i not in seq:
+                        return False
+
+            if self.bitrep_hi:
+                for i in xrange(64):
+                    if not self.bitrep_hi >> i:
+                        break
+                    elif (self.bitrep_hi & (1 << i)) and (i + 64) not in seq:
+                        return False
+        else:
+            dcode, objlen, itemsize, offset, _struct = self.objlist._metadata()
+            for i in xrange(xlen):
+                val = self.objlist._c_getitem(i, dcode, objlen, itemsize, offset, _struct, None)
+                if val not in seq:
+                    return False
+
+        return not strict_subset or xlen < len(seq)
+
+    def __lt__(self, seq):
+        if isinstance(self, proxied_frozenset):
+            return cython.cast(proxied_frozenset, self)._subset(seq, True)
+        else:
+            return cython.cast(proxied_frozenset, seq)._subset(self, True)
+
+    def __le__(self, seq):
+        if isinstance(self, proxied_frozenset):
+            return cython.cast(proxied_frozenset, self)._subset(seq, False)
+        else:
+            return cython.cast(proxied_frozenset, seq)._subset(self, False)
+
+    def __gt__(self, seq):
+        return seq < self
+
+    def __ge__(self, seq):
+        return seq <= self
+
+    def __nonzero__(self):
+        return len(self) > 0
+
+    def __or__(self, seq):
+        return self.union(seq)
+
+    def __ror__(self, seq):
+        return self.union(seq)
+
+    def __and__(self, seq):
+        return self.intersection(seq)
+
+    def __rand__(self, seq):
+        return self.intersection(seq)
+
+    def __sub__(self, seq):
+        return self.difference(seq)
+
+    def __rsub__(self, seq):
+        return frozenset(seq).difference(self)
+
+    def __xor__(self, seq):
+        return self.symmetric_difference(seq)
+
+    def __rxor__(self, seq):
+        return frozenset(seq).symmetric_difference(self)
+
+    def __repr__(self):
+        return "proxied_frozenset(%s)" % str(self)
+
+    def __str__(self):
+        return "[%s]" % ",".join(str(x) for x in self)
```

Comment:
It is wired that you are using the `[]`, since those are used for list. I would put something like `set([...])` that is the value returned by the pure python sets

Reply:
You mean just like the `repr` method ?

Reply:
At least in Python 2, for sets the repr and str returns the same things:

``` 
>>> s = set([1,2,3])
>>> print '%r'% s
set([1, 2, 3])
>>> print '%s' % s
set([1, 2, 3])
```

Reply:
That's good to know. I'll change `__str__`, then.